src/main/java/org/opensearch/neuralsearch/query/ModelInferenceQueryBuilder.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+/**
+ * Query builders which calls ml-commons API to do model inference.
+ * The model inference result is used for search on target field.
+ */
+public interface ModelInferenceQueryBuilder {
+    /**
+     * Get the model id used by ml-commons model inference. Return null if the model id is absent.
+     */
+    public String modelId();
+
+    /**
+     * Set a new model id for the query builder.
+     */
+    public ModelInferenceQueryBuilder modelId(String modelId);
```

Comment:
Is this returned value used anywhere?

Reply:
Yes, it's used by the NeuralSearchQueryVisitor to check whether the model id field is null

Reply:
Can you point the line using this return value?

Reply:
Sure. https://github.com/zhichao-aws/neural-search/blob/1ade82f5566b96fe617ab1f6c5933ebad3258c6d/src/main/java/org/opensearch/neuralsearch/query/visitor/NeuralSearchQueryVisitor.java#L33

Reply:
The line you're pointing at seems the getter method without parameter, am I missing anything?

Reply:
Sorry, I misunderstood your question. The  invokation is here:
https://github.com/zhichao-aws/neural-search/blob/1ade82f5566b96fe617ab1f6c5933ebad3258c6d/src/main/java/org/opensearch/neuralsearch/query/visitor/NeuralSearchQueryVisitor.java#L38
https://github.com/zhichao-aws/neural-search/blob/1ade82f5566b96fe617ab1f6c5933ebad3258c6d/src/main/java/org/opensearch/neuralsearch/query/visitor/NeuralSearchQueryVisitor.java#L40

Reply:
We just set the modelId, and don't use the return value. But this setter method is generated automatically by lombok @Setter annotation. And the return value for setter is the class itself by default.